### PR TITLE
feat: Add extra mounts attribute to HardwareModel

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add functionality, to bind in extra files and folders into your emulation

# Changelog

- Added `get_robot` method to `configuration_file.py` to return robot without having to parse dictionary
- Created Mount models and enums
- In `HardwareModel` defined the `extra-mounts` attribute and the `get_mount_by_name` method
- Added tests to `configuration_file.py`

# Review requests

None

# Risk assessment

Low

### CODE REVIEW

File: emulation_system/emulation_system/compose_file_creator/conversion_logic/conversion_functions.py [hunk 1]
```diff
@@ -50,3 +51,8 @@ def get_image_name_from_hardware_model(hardware_model: HardwareModel) -> Optiona
         hardware_model.emulation_level,
         hardware_model.source_type,
     )
+
+
+def get_bind_mount_string(mount: ExtraMount) -> str:
```
Comment (Laura-Danielle) [inline emulation_system/emulation_system/compose_file_creator/conversion_logic/conversion_functions.py:56]: We can make some of these models use pydantic dataclass support (see [here](https://pydantic-docs.helpmanual.io/usage/dataclasses/)) that way you don't need to write simple helper functions like `get_bind_mount_string` outside of the pydantic model. Instead, you can use dataclass syntax and have this function live inside `ExtraMount`. There might be a few other places you can do this as well.

File: emulation_system/emulation_system/compose_file_creator/input/configuration_file.py [hunk 2]
```diff
@@ -90,6 +90,10 @@ def robot_exists(self) -> bool:
         """Returns True if a robot was defined in config file, False if not."""
         return len(self.robot) > 0
 
+    def get_robot(self) -> Robots:
+        """Returns robot without having to lookup the id."""
+        return self.robot[list(self.robot.keys())[0]]
```
Comment (Laura-Danielle) [inline emulation_system/emulation_system/compose_file_creator/input/configuration_file.py:95]: I am curious as to why the `robot` attribute needs to be a dictionary at all? Especially if we just want to get the ID. And if we want to hold multiple robots inside this class, does it even make sense to have this function at all as opposed to a `get_robot_by_id` function?

**Review by Laura-Danielle — APPROVED** [review]
Overall looks good. Just had two comments.

File: emulation_system/emulation_system/compose_file_creator/input/configuration_file.py (hunk 2 above)
Comment (DerekMaggio) [reply emulation_system/emulation_system/compose_file_creator/input/configuration_file.py:95]: I had it like this to enforce the format of the passed configuration file. Basically it looked cleaner to do it this way.
I am not 100% sure that this was the correct choice. 

Once this whole, "emulation configuration" feature has a working first draft, I plan to come back through and clean it up

File: emulation_system/emulation_system/compose_file_creator/conversion_logic/conversion_functions.py (hunk 1 above)
Comment (DerekMaggio) [reply emulation_system/emulation_system/compose_file_creator/conversion_logic/conversion_functions.py:56]: I did this to promote separation of logic. 
All raw data is put into the pydantic models, any conversion logic is put into the conversion_functions.py file.

Is doing this confusing?

Comment (Laura-Danielle) [reply emulation_system/emulation_system/compose_file_creator/conversion_logic/conversion_functions.py:56]: It's personally a bit confusing for me. My general preference is to give as much control to dataclasses as possible/makes sense to do. That way everything is contained in a single object and I don't have to worry about weird circular dependencies etc to make simply helper functions like `get_bind_mount_string`.

If you think it's easier for you to have them be in separate files, then I think it's fine to keep it as it is. I assume you'll be the one most heavily contributing to this project, so definitely do whatever feels easier to you :)

**Pull request closed without merging**